File: app/src/main/java/com/lifedawn/bestweather/commons/classes/requestweathersource/RequestWeatherSource.java
package com.lifedawn.bestweather.commons.classes.requestweathersource;

import com.lifedawn.bestweather.data.remote.retrofit.client.RetrofitClient;

import java.util.HashSet;
import java.util.Set;

public class RequestWeatherSource {
	private Set<RetrofitClient.ServiceType> requestServiceTypes = new HashSet<>();

	public Set<RetrofitClient.ServiceType> getRequestServiceTypes() {
		return requestServiceTypes;
	}

	public RequestWeatherSource addRequestServiceType(RetrofitClient.ServiceType serviceType) {
		requestServiceTypes.add(serviceType);
		return this;
	}
}

File: app/src/main/java/com/lifedawn/bestweather/commons/classes/requestweathersource/RequestWeatherSource.kt
package com.lifedawn.bestweather.commons.classes.requestweathersource

import com.lifedawn.bestweather.data.remote.retrofit.client.RetrofitClient.ServiceType

open class RequestWeatherSource {
    private val requestServiceTypes: MutableSet<ServiceType> = HashSet()
    fun getRequestServiceTypes(): Set<ServiceType> {
        return requestServiceTypes
    }

    fun addRequestServiceType(serviceType: ServiceType): RequestWeatherSource {
        requestServiceTypes.add(serviceType)
        return this
    }
}